### PR TITLE
Blog with google

### PR DESCRIPTION
**Pull request by gussiciliano**
### Sumary
- Add authentication with Gmail.

### Trello Card
https://trello.com/c/7vDciWH2/15-integrar-el-blog-con-google-login

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -8,6 +8,16 @@
   # by default. You can change it below and use your own secret key.
   # config.secret_key = '815ec6378bc130a445d4c52206cca72c1b5214b626fc0edbf9f61d6910f2c858189e77827d9d485504c8b22c87e35d24965c8bae0058aa4332e98bdcb9524dc8'
 
+  config.omniauth :google_oauth2, 
+                  "64051397506-ieqkt5849a9sk0ros4a4pstec17a5tt8.apps.googleusercontent.com",
+                  "kAPA1_g7luyFYDkISv3_I2-s",
```
Comment (Dinuuu) [inline config/initializers/devise.rb]: this keys shouldn't be pushed into the repository

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -8,6 +8,16 @@
   # by default. You can change it below and use your own secret key.
   # config.secret_key = '815ec6378bc130a445d4c52206cca72c1b5214b626fc0edbf9f61d6910f2c858189e77827d9d485504c8b22c87e35d24965c8bae0058aa4332e98bdcb9524dc8'
 
+  config.omniauth :google_oauth2, 
+                  "64051397506-ieqkt5849a9sk0ros4a4pstec17a5tt8.apps.googleusercontent.com",
```
Comment (Dinuuu) [inline config/initializers/devise.rb]: idem

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -2,6 +2,19 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   # You should configure your model like this:
   # devise :omniauthable, omniauth_providers: [:twitter]
 
+  def google_oauth2
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      flash[:notice] = I18n.t "devise.omniauth_callbacks.success", :kind => "Google"
```
Comment (Dinuuu) [inline app/controllers/users/omniauth_callbacks_controller.rb]: remember to use the new hash syntax `key: value`

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -2,6 +2,19 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
...
+      sign_in_and_redirect @user, :event => :authentication
```
Comment (Dinuuu) [inline app/controllers/users/omniauth_callbacks_controller.rb]: idem

File: app/models/user.rb [hunk 5]
```diff
@@ -16,4 +17,17 @@ def generate_verification_code
     self.verification_code = AuthenticableEntity.verification_code
   end
   # ...
+
+  def self.from_omniauth(access_token)
+    data = access_token.info
+    user = User.where(:email => data["email"]).first
```
Comment (Dinuuu) [inline app/models/user.rb]: idem

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 3 above)
Comment (gussiciliano) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Oks

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 4 above)
Comment (gussiciliano) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Oks

File: app/models/user.rb (hunk 5 above)
Comment (gussiciliano) [inline app/models/user.rb]: Oks

**Pull request merged**